src/lib.rs

```diff
@@ -233,6 +237,10 @@ decl_module! {
             ensure!(<EnclaveIndex<T>>::contains_key(&sender),
             "[Teerex]: IPFS state update requested by enclave that is not registered");
 
+            let sender_index = <EnclaveIndex<T>>::get(sender);
+            ensure!(<EnclaveRegistry::<T>>::get(sender_index).mr_enclave.encode() == bonding_account.encode(),
+            "[Teerex]: IPFS state update requested by enclave that doesn't match the bonding accounts");
```

Comment:
I would like to have new error variants introduced here. String errors miss information when they are returned client side.

Reply:
Please do that for all string errors that are returned in the pallet.